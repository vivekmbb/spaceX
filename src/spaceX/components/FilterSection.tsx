import React from "react";
import { Box, Text } from "grommet";

export const FilterSection = ({ props }: any) => {
  const { selectYear, setSelectYear, selectLaunch, setSelectLaunch, selectLanding, setSelectLanding } = props;
  const years = [
    ["2006", "2007"],
    ["2008", "2009"],
    ["2010", "2011"],
    ["2012", "2013"],
    ["2014", "2015"],
    ["2016", "2017"],
    ["2018", "2019"],
    ["2020"]
  ];

  const launch = [["True", "False"]];

  const landing = [["True", "False"]];

  const getFilter = (clickedInput: any, setClickedInput: any, list: any) => {
    return React.Children.toArray(
      list?.map((object: any, index: number) => (
        <Box direction="row" gap="xlarge" key={index}>
          <Box
            align="start"
            background={clickedInput === object[0] ? "neutral-1" : "accent-1"}
            justify="center"
            onClick={() => setClickedInput(object[0] === clickedInput ? undefined : object[0])}
            pad="small"
            round="6px"
            margin="5px"
          >
            {object[0]}
          </Box>
          {object.length === 2 && (
            <Box
              align="end"
              justify="center"
              background={clickedInput === object[1] ? "neutral-1" : "accent-1"}
              onClick={() => setClickedInput(object[1] === clickedInput ? undefined : object[1])}
              pad="small"
              round="6px"
              margin="5px"
            >
              {object[1]}
            </Box>
          )}
        </Box>
      ))
    );
  };

  return (
    <Box background="white" round="8px" pad={{ bottom: "medium" }}>
      <Text size="medium" weight="bold" margin={{ left: "medium", top: "small" }}>
        Filters
      </Text>
      <Box align="center" border={{ side: "bottom", size: "1px" }} margin={{ left: "large", right: "large" }}>
        <Text size="medium" margin={{ bottom: "small" }}>
          Launch Year
        </Text>
      </Box>
      <Box data-testid="button section" margin="auto">
        {getFilter(selectYear, setSelectYear, years)}
      </Box>
      <Box
        align="center"
        border={{ side: "bottom", size: "1px" }}
        margin={{ left: "large", right: "large", top: "medium" }}
      >
        <Text size="medium" margin={{ bottom: "small" }}>
          Successful Launch
        </Text>
      </Box>
      <Box data-testid="button section" margin="auto">
        {getFilter(selectLaunch, setSelectLaunch, launch)}
      </Box>
      <Box
        align="center"
        border={{ side: "bottom", size: "1px" }}
        margin={{ left: "large", right: "large", top: "medium" }}
      >
        <Text size="medium" margin={{ bottom: "small" }}>
          Successful Landing
        </Text>
      </Box>
      <Box data-testid="button section" margin="auto">
        {getFilter(selectLanding, setSelectLanding, landing)}
      </Box>
    </Box>
  );
};
