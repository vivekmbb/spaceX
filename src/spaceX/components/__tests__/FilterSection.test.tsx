// tslint:disable: no-submodule-imports
import "@testing-library/jest-dom/extend-expect";
import { Box } from "grommet";
import React from "react";
import renderer from "react-test-renderer";
import { FilterSection } from "../FilterSection";

describe("Filter Section Component", () => {
  it("Filter Section component", () => {
    const Wrapper = (props: any) => <Box pad="medium">{props.children}</Box>;

    const ChildFunction = () => {
      const [selectYear, setSelectYear] = React.useState();
      const [selectLaunch, setSelectLaunch] = React.useState();
      const [selectLanding, setSelectLanding] = React.useState();

      const props = {
        selectYear,
        setSelectYear,
        selectLaunch,
        setSelectLaunch,
        selectLanding,
        setSelectLanding
      };

      return <FilterSection props={props} />;
    };
    const component = renderer
      .create(
        <Wrapper>
          <ChildFunction />
        </Wrapper>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
