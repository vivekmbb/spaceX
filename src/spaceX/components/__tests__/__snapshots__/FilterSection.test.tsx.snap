// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Section Component Filter Section component 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 chkvLT"
>
  <div
    className="StyledBox-sc-13pk1d4-0 irFSzw"
  >
    <span
      className="StyledText-sc-1sadyjn-0 eeaybW"
    >
      Filters
    </span>
    <div
      className="StyledBox-sc-13pk1d4-0 fPUIqt"
    >
      <span
        className="StyledText-sc-1sadyjn-0 toSoo"
      >
        Launch Year
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fCcErq"
      data-testid="button section"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 iSmarO"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 IjQKZ"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2006
        </div>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 kxsFh"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 kiMtHO"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2007
        </div>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 iSmarO"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 IjQKZ"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2008
        </div>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 kxsFh"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 kiMtHO"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2009
        </div>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 iSmarO"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 IjQKZ"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2010
        </div>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 kxsFh"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 kiMtHO"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2011
        </div>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 iSmarO"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 IjQKZ"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2012
        </div>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 kxsFh"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 kiMtHO"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2013
        </div>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 iSmarO"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 IjQKZ"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2014
        </div>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 kxsFh"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 kiMtHO"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2015
        </div>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 iSmarO"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 IjQKZ"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2016
        </div>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 kxsFh"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 kiMtHO"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2017
        </div>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 iSmarO"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 IjQKZ"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2018
        </div>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 kxsFh"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 kiMtHO"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2019
        </div>
      </div>
      <div
        className="StyledBox-sc-13pk1d4-0 iSmarO"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 IjQKZ"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          2020
        </div>
      </div>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 dpWyCI"
    >
      <span
        className="StyledText-sc-1sadyjn-0 toSoo"
      >
        Successful Launch
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fCcErq"
      data-testid="button section"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 iSmarO"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 IjQKZ"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          True
        </div>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 kxsFh"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 kiMtHO"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          False
        </div>
      </div>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 dpWyCI"
    >
      <span
        className="StyledText-sc-1sadyjn-0 toSoo"
      >
        Successful Landing
      </span>
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 fCcErq"
      data-testid="button section"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 iSmarO"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 IjQKZ"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          True
        </div>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 kxsFh"
        />
        <div
          className="StyledBox-sc-13pk1d4-0 kiMtHO"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          tabIndex={0}
        >
          False
        </div>
      </div>
    </div>
  </div>
</div>
`;
